Fix typo that caused use of glyph 0 sometimes.

zapolnov opened the pull request:
There is a code that, as I assume, should check if glyph is 0 (which generally means "no glyph"), but instead it checks variable `glyph`. That variable is a pointer and it is unconditionally referenced a line before, so it never could be a null pointer.

This typo causes incorrect rendering at least with the Jozoor Arabic Font (https://fonts.jozoor.com/jozoor-font). After the fix it renders correctly.

behdad commented:
Ah. Thanks.

behdad commented:
I'm curious.  In what kind of environment / application are you using icu-le-hb?

zapolnov commented:
In a mobile game. We was using ICU's layout engine, but it has been deprecated recently and removed from ICU distribution. So after update of ICU we have switched to icu-le-hb, as suggested by the ICU's changelog.

behdad commented:
Cool.  Good to know people actually are using this.
